docker/README.md

```diff
@@ -0,0 +1,61 @@
+# Docker build image
+
+This directory contains a docker file for an ESPurna build environment. All docker commands are intended to be run from your main source repository directory.
+
+Two volumes can be used. 
+* `/espurna` with ESPurna source code.
+* `/firmware` target directory for complied firmware files.
+
+Using the commands below, files would be output to `/tmp/firmware/` on your docker host system.
+
+## Build
+
+```bash
+# from the main repo directory espurna/
+sudo docker build -t espurna-build -f docker/Dockerfile .
+```
+
+## Examples
+
+### Listing Firmware Images
+```bash
+sudo docker run --rm -it -v /tmp/firmware/:/firmware espurna-build -l
+--------------------------------------------------------------
+Available environments:
+* aithinker-ai-light
+* allnet-4duino-iot-wlan-relais
+* allterco-shelly1
+* allterco-shelly1pm
+* allterco-shelly2
+...
+```
+
+### Building All Firmware
+This simple example will build all firmware files from dev branch to /tmp/firmware.
+
+```bash
+sudo docker run --rm -it -v /tmp/firmware/:/firmware espurna-build
+```
+
+### Building A Single Environment
+This example will only build firmware for environment `intermittech-quinled` from local files in `/home/user/espurna`
+
+```bash
+udo docker run --rm -it -v /tmp/firmware/:/firmware espurna-build intermittech-quinled
+--------------------------------------------------------------
+ESPURNA FIRMWARE BUILDER
+Building for version 1.15.0-dev (83a49b19)
+--------------------------------------------------------------
+Building web interface...
+[18:38:22] Using gulpfile /espurna/code/gulpfile.js
+[18:38:22] Starting 'default'...
+[18:38:22] Starting 'webui'...
+...
+...
+Cloning into '/root/.platformio/.cache/tmp/pkg-installing-rwukc302'...
+Cloning into '/root/.platformio/.cache/tmp/pkg-installing-20osjkr2'...
```

Comment:
I think previous version meant to use userid map here?

Reply:
previously, but there was questions on if everyone would have a valid userid that started at 1000 or similar, figured it was just easier to drop it than deal with it as most times people just sudo the docker anyway (for good or bad).

Reply:
If this is just an example. Build sends things as root already, so the file needs some tweaking too, i.e. docker run --id is not enough.

Reply:
I don't know that I understand - are you wanting to wire back in the previous ID stuff that wasn't working / might not be compatible or you are looking for something else?

Reply:
Previous ID was fixed, true, I'm just saying that in case docker run --id=$(id -u) is used, it will fetch build tools all over again, since PIO installs stuff in the $HOME/.platformio

Maybe we want a comment in the readme with --build-arg UID=$(id -u) --build-arg GID=$(id -g)?
https://docs.docker.com/engine/reference/commandline/build/#set-build-time-variables---build-arg